tests/acceptance/01_vars/01_basic/os_version_major.cf

```diff
@@ -10,105 +16,52 @@ bundle agent test
         string => "Test for expected values of variable sys.os_version_major";
 
   vars:
-    any::
-      "actual" string => "$(sys.os_version_major)";
-      "expected" string => "Unknown";
-    debian_7::
-      "expected" string => "7";
-    debian_8::
-      "expected" string => "8";
-    debian_9::
-      "expected" string => "9";
-    debian_10::
-      "expected" string => "10";
-
-    ubuntu_14::
-      "expected" string => "14";
-    ubuntu_16::
-      "expected" string => "16";
-    ubuntu_18::
-      "expected" string => "18";
-    ubuntu_20::
-      "expected" string => "20";
-
-    redhat_5::
-      "expected" string => "5";
-    redhat_6::
-      "expected" string => "6";
-    redhat_7::
-      "expected" string => "7";
-    redhat_8::
-      "expected" string => "8";
-
-    centos_6::
-      "expected" string => "6";
-    centos_7::
-      "expected" string => "7";
-    centos_8::
-      "expected" string => "8";
-
-    fedora_32::
-      "expected" string => "32";
-    fedora_33::
-      "expected" string => "33";
+      # Platforms to test
+      "platforms"
+        slist => { "debian", "ubuntu", "redhat", "rhel", "centos", "fedora",
+                   "aix", "hpux", "suse", "opensuse", "sles", "solaris",
+                   "sunos", "windows", "freebsd", "macos" };
 
-    aix_5::
-      "expected" string => "5";
-    aix_6::
-      "expected" string => "6";
-    aix_7::
-      "expected" string => "7";
-    aix_8::
-      "expected" string => "8";
+      # Regex matching current platforms OS-class with version numbers
+      "class_regex"
+        string => format("^(%s)_([a-z]_)?[0-9]+_[0-9]+$", join("|", "platforms"));
 
-    hpux_10::
-      "expected" string => "10";
-    hpux_11::
-      "expected" string => "11";
+      # Regex to extract major version number from OS-class
+      # Edge cases:
+      #  - On Solaris/SunOS major version comes second
+      #    E.g. Solaris 11 has class "solaris_5_11"
+      #  - On HP-UX there is an additional model letter
+      #    E.g. HP-UX 11 has class "hpux_b_11_23"
+      "extract_regex"
+        string => ifelse("solaris|sunos", "^[a-z]+_[0-9]+_([0-9]+$)",
+                         "hpux",          "^[a-z]+_[a-z]_([0-9]+)_[0-9]+$",
```

Comment:
@larsewi I am assuming that `hpux` is actually defined - if not we have a problem.

Then, according to this test `hpux_11` should also be defined:

https://github.com/cfengine/core/blob/86079af65b1f6a7b4069d0a07837e4fcb712a3cc/tests/acceptance/02_classes/01_basic/hp_ux_major.cf

Maybe your regex is too generic and matching other things? What if it was just:

```
hpux_([0-9]+)$
```

Reply:
Just checked, both `hpux` and `hpux_11` are defined. No need to worry. On the other hand `hpux_b_11_23` was not defined, but `hp_ux_b_11_23` was defined :wink: